File: tools/cpp/tools_test.cpp
#include "tools.h"
#include <cassert>
#include <iostream>

void test_echo() {
    auto result = agent_tools::echo("hello");
    assert(result.success);
    assert(result.output == "hello");
    assert(result.error.empty());
    std::cout << "PASS: test_echo\n";
}

void test_echo_ffi() {
    char buf[256];
    int n = echo_ffi("hello ffi", buf, sizeof(buf));
    assert(n > 0);
    assert(std::string(buf) == "hello ffi");
    std::cout << "PASS: test_echo_ffi\n";
}

int main() {
    test_echo();
    test_echo_ffi();
    std::cout << "All tests passed.\n";
    return 0;
}

File: tools/go/tools_test.go
package tools

import (
	"context"
	"testing"
)

func TestEcho(t *testing.T) {
	ctx := context.Background()

	tests := []struct {
		name       string
		input      string
		wantOutput string
		wantOK     bool
	}{
		{"non-empty input returned unchanged", "hello world", "hello world", true},
		{"empty string returned unchanged", "", "", true},
		{"success is true", "any", "any", true},
	}

	for _, tc := range tests {
		t.Run(tc.name, func(t *testing.T) {
			got, err := Echo(ctx, Args{Data: tc.input})
			if err != nil {
				t.Fatalf("Echo returned unexpected error: %v", err)
			}
			if got.Output != tc.wantOutput {
				t.Errorf("Output = %q, want %q", got.Output, tc.wantOutput)
			}
			if got.Success != tc.wantOK {
				t.Errorf("Success = %v, want %v", got.Success, tc.wantOK)
			}
		})
	}
}
